### PR TITLE
Added hardhat deploy script to easily deploy all contracts

### PR DESCRIPTION
**Pull request by schystz**
Adds a hardhat deploy script for deploying the Rewards contract and other dependency contracts

### CODE REVIEW

File: hardhat/scripts/deploy.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ammLpRewardsRatio, //in bps, multiplied by 10^4
+    vestingRewardsRatio, //in bps, multiplied by 10^4
+    minterContract.address,
+    genesisTs,
+    minterLpPools,
+    ammLpPools
+  )
+  await rewardsContract.deployed()
+  console.log('rewardsContract deployed at: ', rewardsContract.address)
+}
```
Comment (ernestpascual) [inline hardhat/scripts/deploy.js:76]: @schystz  we need to transfer some halo to the rewards contract. we can add it after deploying the rewards contract. Right now I'm minting new tokens directly to the rewards contract:

```
  const INITIAL_MINT = 10 ** 6;
  
  await haloTokenContract.mint(
    rewardsContract.address,
    ethers.utils.parseEther((40 * INITIAL_MINT).toString())
  );
```

 I have this now. I got this from the rewards test. (https://github.com/HaloDAO/halo-rewards/blob/8d32983f0ea613b81a17ae0ed885bd0f1843ac5a/hardhat/test/reward-test.js#L17)

File: hardhat/scripts/deploy.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   */
+  const startingRewards = ethers.utils.parseEther('7500000')
+  const decayBase = ethers.utils.parseEther('0.813')
+  const epochLength = 60
+  const minterLpRewardsRatio = 0.4 * BPS
+  const ammLpRewardsRatio = 0.4 * BPS
+  const vestingRewardsRatio = 0.2 * BPS
+  const genesisTs = Math.floor(Date.now() / 1000)
+  const minterLpPools = [[collateralERC20Contract.address, 10]]
+  const ammLpPools = [[lpTokenContract.address, 10]]
```
Comment (ernestpascual) [inline hardhat/scripts/deploy.js:58]: Do we hardcode the current balancer pool address that we have now? cc: @tracyarciaga @ctverceles

File: hardhat/scripts/deploy.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+   * Deploy dummy contracts (required by Rewards contract)
+   * - collateral token
+   * - LP token contract
+   * - minter
+   */
+  const CollateralERC20 = await ethers.getContractFactory('CollateralERC20')
+  const collateralERC20Contract = await CollateralERC20.deploy('Dai', 'DAI')
+  await collateralERC20Contract.deployed()
+
+  const LpToken = await ethers.getContractFactory('LpToken')
```
Comment (ernestpascual) [inline hardhat/scripts/deploy.js:33]: Thinking of removing this and just add the balancer tokens instead.

You can find it on https://kovan.pools.balancer.exchange/#/pool/0x37f80ac90235ce0d3911952d0ce49071a0ffdb1e/about

**Review by ernestpascual — CHANGES_REQUESTED** [review]
Left some minor comments in the deploy script @schystz Looks good overall!

File: hardhat/scripts/deploy.js (hunk 3 above)
Comment (schystz) [reply hardhat/scripts/deploy.js:33]: Agree, let me update this!

File: hardhat/scripts/deploy.js (hunk 2 above)
Comment (schystz) [reply hardhat/scripts/deploy.js:58]: Yeah let's just hardcode the kovan balancer pools

File: hardhat/scripts/deploy.js (hunk 1 above)
Comment (schystz) [reply hardhat/scripts/deploy.js:76]: Thanks for the catch, will add this!

**Review by tracyarciaga — APPROVED** [review]

**Comment by schystz** [comment]
@ernestpascual updated with your suggestions! Have added 2 balancer pool addresses for the initial LP contracts.

File: hardhat/scripts/deploy.js (hunk 3 above)
Comment (schystz) [reply hardhat/scripts/deploy.js]: Removed

File: hardhat/scripts/deploy.js (hunk 2 above)
Comment (schystz) [reply hardhat/scripts/deploy.js]: Added 2 kovan balancer pool addresses

File: hardhat/scripts/deploy.js (hunk 1 above)
Comment (schystz) [reply hardhat/scripts/deploy.js:84]: Added the initial mint

**Review by ernestpascual — APPROVED** [review]
looking good. thanks james! ready to merge

**Comment by ernestpascual** [comment]
Sorry, one more thing before merging. Can we add this in package.json "deploy:kovan": "npx hardhat run scripts/deploy.js --network \"kovan\""

this is the command for using the deploy script to deploy to kovan. thanks! @schystz

**Comment by schystz** [comment]
@ernestpascual added the `deploy:kovan` command. Pls let me know if it looks good to you!

**Review by ernestpascual — APPROVED** [review]
thanks james!

**Pull request merged**